core/store/src/trie/mem/updating.rs

```diff
@@ -519,10 +519,16 @@ impl<'a, M: ArenaMemory> MemTrieUpdate<'a, M> {
         }
     }
 
-    fn add_refcount_to_value(&mut self, hash: CryptoHash, value: Option<Vec<u8>>) {
-        if let Some(tracked_node_changes) = self.tracked_trie_changes.as_mut() {
-            tracked_node_changes.refcount_changes.add(hash, value.unwrap(), 1);
-        }
+    // We change refcount only when we also make disk updates.
+    // In this case, we must have Full value.
+    fn add_refcount_to_value(&mut self, value: ValueToInsert) {
```

Comment:
It feels like the determination of whether to add refcount should be made based on a boolean that says "do I need disk updates", not whether the value to insert happens to be Full? I don't think the original interface was great either, but this kind of behavior feels implicit. Eh... but I guess I don't see a cleaner approach.

Reply:
Made naming more explicit: `MemtrieAndDisk/MemtrieOnly`